app/models/katello/glue/pulp/node.rb

```diff
@@ -0,0 +1,23 @@
+module Katello
+  module Glue::Pulp::Node
+    def self.included(base)
+      base.send :include, InstanceMethods
+      base.send :include, LazyAccessor
+
+      base.class_eval do
+        lazy_accessor :pulp_repositories, :initializer => lambda { |_s| pulp_node.extensions.repository.retrieve_all }
+      end
+    end
+
+    module InstanceMethods
+
+      def pulp_node
+        @pulp_node ||= User.runcible_instance(pulp_url, User.remote_user)
+      end
```

Comment:
I don't think this will work today with the authentication we use for capsules because of the oauth being different than the pulp server installed on the satellite itself.  There is some work going on as part of the pulp 2.8 work to support using cert based auth to talk to the capsules.